### PR TITLE
resolves #1019 : Orchestration timeout for conversation with no inter…

### LINKED ISSUE #1019: [Orchestration] Defined timeout for an orchestration without interaction
Possibility for the primary bot to take back delegation when there is no conversation for a while

### PR DESCRIPTION
**Pull request by delphes99**
Orchestration timeout for conversation with no interaction for a while

Signed-off-by: Laurent Gautho-lapeyre <mail@github.laurent-gautho-lapeyre.fr>

### CODE REVIEW

File: bot/orchestration/src/main/kotlin/ai/tock/bot/orchestration/bot/primary/RunOrchestrationStoryListener.kt [hunk 1]
```diff
@@ -147,25 +148,29 @@ class RunOrchestrationStoryListener(
...
             if (intent.inNoOrchestrationList()) {
                 logger.info { "End of the orchestration caused by the ${intent?.wrappedIntent()?.name ?: "???"} intent" }
                 orchestrationRepository.end(orchestration.playerId)
 
-                send("Your conversation with ${orchestration.targetBot.botLabel} is now over.")
-                return true
+                return configuration.onNoOrchestration(this, orchestration) == ComeBackFromSecondary.EXECUTE_INITIAL_STORY
+            }
+
+            if(configuration.isTimeout != null && userTimeline.currentDialog?.lastUserAction?.date?.let(configuration.isTimeout) == true) {
```
Comment (delphes99) [inline bot/orchestration/src/main/kotlin/ai/tock/bot/orchestration/bot/primary/RunOrchestrationStoryListener.kt:169]: je me posais la question est ce que je devrais pas plutôt faire un truc beaucoup plus générique du style "configuration.stopOrcherstation(bus)"...

File: bot/orchestration/src/main/kotlin/ai/tock/bot/orchestration/bot/primary/RunOrchestrationStoryListener.kt [hunk 2]
```diff
@@ -147,25 +148,29 @@ class RunOrchestrationStoryListener(
         }
     }
 
-    private fun BotBus.resumeOrchestration(orchestration: Orchestration): Boolean {
+    private fun BotBus.resumeOrchestration(orchestration: Orchestration, userTimeline: UserTimeline): Boolean {
```
Comment (curinator) [inline bot/orchestration/src/main/kotlin/ai/tock/bot/orchestration/bot/primary/RunOrchestrationStoryListener.kt:151]: `userTimeline` is already embedded in `this` here

File: bot/orchestration/src/main/kotlin/ai/tock/bot/orchestration/bot/primary/DefaultPrimaryBotOrchestrationEventHandler.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package ai.tock.bot.orchestration.bot.primary
+
+import ai.tock.bot.definition.StoryDefinition
+import ai.tock.bot.engine.BotBus
+import ai.tock.bot.orchestration.shared.AvailableOrchestrationResponse
+
+open class DefaultPrimaryBotOrchestrationEventHandler(private val comebackStory: StoryDefinition) : PrimaryBotOrchestrationEventHandler {
+    override fun onStarOrchestration(bus: BotBus, orchestrationResponse: AvailableOrchestrationResponse) {
+        bus.send("Pour vous répondre je passe la main à mon collègue ${orchestrationResponse.targetBot.botLabel} !")
```
Comment (curinator) [inline bot/orchestration/src/main/kotlin/ai/tock/bot/orchestration/bot/primary/DefaultPrimaryBotOrchestrationEventHandler.kt:9]: @vsct-jburet  do we want to keep this french sentence ?

Comment (vsct-jburet) [reply bot/orchestration/src/main/kotlin/ai/tock/bot/orchestration/bot/primary/DefaultPrimaryBotOrchestrationEventHandler.kt:9]: no !:)

File: bot/orchestration/src/main/kotlin/ai/tock/bot/orchestration/bot/primary/PrimaryBotConfiguration.kt [hunk 4]
```diff
@@ -22,16 +22,27 @@ import ai.tock.bot.definition.StoryDefinition
...
 ) {
-    fun getEligibleTargetBots(connectorType: ConnectorType) : List<OrchestrationTargetedBot> = eligibleTargetBotsByConnector[connectorType] ?: emptyList()
+    fun getEligibleTargetBots(connectorType: ConnectorType): List<OrchestrationTargetedBot> =
+        eligibleTargetBotsByConnector[connectorType] ?: emptyList()
 
-    fun getOrchestrationData(bus : BotBus) : OrchestrationData? = dataProvider.provideOrchestrationData(bus)
+    fun getOrchestrationData(bus: BotBus): OrchestrationData? = dataProvider.provideOrchestrationData(bus)
+
+    companion object {
+        fun takeBackOrchestrationByTimeOut(inactivityDuration: Duration) : ((BotBus) -> Boolean) {
```
Comment (delphes99) [inline bot/orchestration/src/main/kotlin/ai/tock/bot/orchestration/bot/primary/PrimaryBotConfiguration.kt:44]: I have added this example function here but I don't know if it's the right place

**Pull request merged**